src/modules/collection/sagas.spec.ts

```diff
@@ -2377,3 +2380,163 @@ describe('when handling a deploy missing entities', () => {
+        rarity: 'common' as any, // Added rarity to prevent toLowerCase error
+        price: '100000000000000000000' // 100 MANA
+      })
+    ]
+    email = 'test@example.com'
+    subscribeToNewsletter = false
+    from = '0x' + '123'.padStart(40, '0')
+    txHash = '0xdeadbeef'
+    creditsServerUrl = 'https://credits.example.com'
+    creditsResponse = {
+      credits: [
+        {
+          id: 'credit-1',
+          amount: '1000000000000000000', // 1 MANA
+          availableAmount: '1000000000000000000',
+          expiresAt: '9999999999',
+          signature: '0xsignature',
+          contract: '0x5309ae874fc4eb21adcd63f8b6c3f766cc3b1849',
+          season: 1,
+          timestamp: '1234567890',
+          userAddress: from
+        }
+      ],
+      totalCredits: 1000000000000000000
+    }
+  })
+
+  describe('when using credits with MANA payment method', () => {
+    let originalFetch: typeof global.fetch
+    let useCreditsCollectionManagerSpy: jest.SpyInstance
+
+    beforeEach(() => {
+      ;(mockBuilder.fetchCollectionItems as jest.Mock).mockResolvedValue(items)
+      // Mock fetch globally for credits server signature
+      originalFetch = global.fetch
+      global.fetch = jest.fn().mockResolvedValue({
+        ok: true,
+        json: () => Promise.resolve({ signature: '0xexternalCallSignature', messageHash: '0xhash' } as any)
+      } as any)
+      // Mock useCreditsCollectionManager to return txHash directly
+      useCreditsCollectionManagerSpy = jest.spyOn(CreditsService.prototype, 'useCreditsCollectionManager').mockResolvedValue(txHash)
+    })
+
+    afterEach(() => {
+      // Restore original fetch
+      global.fetch = originalFetch
+      // Restore useCreditsCollectionManager
+      useCreditsCollectionManagerSpy.mockRestore()
+    })
+
+    it('should successfully publish the collection using credits', () => {
+      const rarities: BlockchainRarity[] = [
+        {
+          id: 'common',
+          name: 'Common',
+          price: '100000000000000000000',
+          prices: {
+            MANA: '100000000000000000000'
+          },
+          maxSupply: '1000000000'
+        } as unknown as BlockchainRarity
+      ]
+
+      return expectSaga(collectionSaga, mockBuilder, mockBuilderClient)
+        .provide([
+          [put(saveCollectionRequest(collection)), true],
+          [
+            race({ success: take(SAVE_COLLECTION_SUCCESS), failure: take(SAVE_COLLECTION_FAILURE) }),
+            { success: saveCollectionSuccess(collection) }
+          ],
+          [call([mockBuilder, 'fetchCollectionItems'], collection.id), items],
+          [select(getAddress), from],
+          [call(getChainIdByNetwork, Network.MATIC), ChainId.MATIC_AMOY],
+          [retry(10, 500, mockBuilder.saveTOS, TermsOfServiceEvent.PUBLISH_COLLECTION, collection, email), undefined],
+          [put(fetchRaritiesRequest()), undefined],
```

Comment:
With the proposed changes, we can mock the `CreditsService.prototype, useCreditsCollectionManager` here and make it return what we want. We can also test that it's called with the correct parameters.